### PR TITLE
Few makefile/readme updates

### PR DESCRIPTION
**Pull request by atishp04**
- Add busybox(using buildroot) rootfs option
- Bring back hvc0 (this should be only merged after the [kernel patch](https://github.com/rivosinc/linux/pull/1) is merged)

### CODE REVIEW

**Review by glg-rv — COMMENTED** [review]
Just minor comments

File: README.md [hunk 1]
```diff
@@ -34,7 +34,12 @@ Linux kernel:
 - Out-of-tree patches are required; see table below.
 - Build: `ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- make defconfig Image`
 - Set the `LINUX=` variable to point to the compiled Linux kernel tree when
-  using the `make run_linux` and `make run_debian` targets described below.
+  using linux related targets described below.
+
+Buildroot:
+- Out-of-tree patches are required; see table below.
+- Build `make qemu_riscv64_virt_defconfig;make`
+- Set the `BUILDROOT=` variable to point to the buildroot source directory.
```
[the remark this answers is not in the document]
Comment (atishp04) [reply README.md]: Done

File: README.md [hunk 2]
```diff
@@ -34,7 +34,12 @@ Linux kernel:
 - Out-of-tree patches are required; see table below.
 - Build: `ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- make defconfig Image`
 - Set the `LINUX=` variable to point to the compiled Linux kernel tree when
-  using the `make run_linux` and `make run_debian` targets described below.
+  using linux related targets described below.
+
+Buildroot:
+- Out-of-tree patches are required; see table below.
+- Build `make qemu_riscv64_virt_defconfig;make`
```
[the remark this answers is not in the document]
Comment (atishp04) [reply README.md]: Fixed.

File: Makefile [hunk 3]
```diff
@@ -132,6 +135,32 @@ run_debian: salus
 		-netdev user,id=usernet,hostfwd=tcp:127.0.0.1:7722-0.0.0.0:22 \
 		-device e1000e,netdev=usernet \
 		$(EXTRA_QEMU_ARGS)
+run_buildroot: salus
```
[the remark this answers is not in the document]
Comment (atishp04) [reply Makefile:140]: Yeah. I have moved the common qemu arguments to a variable. PTAL.

File: Makefile [hunk 4]
```diff
@@ -129,24 +132,21 @@ run_debian: salus
...
+		-$(IOMMU_ARGS) \
+		-$(NETWORK_ARGS) \
 		$(EXTRA_QEMU_ARGS)
 run_buildroot: salus
 	$(QEMU_BIN) \
 		$(MACH_ARGS) \
 		-kernel $(RELEASE_BINS)salus \
 		-device guest-loader,kernel=$(LINUX_BIN),addr=$(KERNEL_ADDR) \
 		-append "$(BUILDROOT_BOOTARGS)" \
-		-drive file="$(BUILDROOT_INITRD_IMAGE),format=raw,id=hd0" \
```
Comment (dgreid) [inline Makefile]: s/hd0/hd/ is not mentioned in the commit message. Should that be in a prior commit?

File: Makefile [hunk 5]
```diff
@@ -132,6 +135,32 @@ run_debian: salus
...
+		$(MACH_ARGS) \
+		-kernel $(RELEASE_BINS)salus \
+		-device guest-loader,kernel=$(LINUX_BIN),addr=$(KERNEL_ADDR) \
+		-append "$(BUILDROOT_BOOTARGS)" \
+		-drive file="$(BUILDROOT_INITRD_IMAGE),format=raw,id=hd0" \
+		-device x-riscv-iommu-pci \
+		-device nvme,serial=deadbeef,drive=hd0 \
+		-netdev user,id=usernet,hostfwd=tcp:127.0.0.1:7722-0.0.0.0:22 \
+		-device e1000e,netdev=usernet \
+		-d unimp -D log.txt\
```
[the remark this answers is not in the document]
Comment (atishp04) [reply Makefile]: Oops. I removed them in PATCH 4. Will update this patch. Thanks for catching.

File: Makefile [hunk 6]
```diff
@@ -22,13 +22,15 @@ LINUX_BIN := $(LINUX)/arch/riscv/boot/Image
 DEBIAN ?=
 ROOTFS_IMAGE := $(DEBIAN)/image.qcow2
 INITRD_IMAGE := $(DEBIAN)/initrd
+BUILDROOT_INITRD_IMAGE := $(BUILDROOT)/output/images/rootfs.ext2
```
[the remark this answers is not in the document]
Comment (atishp04) [reply Makefile:25]: Done.

File: Makefile [hunk 7]
```diff
@@ -31,6 +31,9 @@ KERNEL_ADDR := 0xc0200000
 INITRD_ADDR := 0xc2200000
```
[the remark this answers is not in the document]
Comment (atishp04) [reply Makefile:31]: Done.

File: Makefile (hunk 4 above)
Comment (atishp04) [reply Makefile]: Squashed the PATCH 1 & 4 now.

File: Makefile [hunk 8]
```diff
@@ -22,13 +22,18 @@ LINUX_BIN := $(LINUX)/arch/riscv/boot/Image
...
 INITRD_IMAGE := $(DEBIAN)/initrd
+BUILDROOT_IMAGE := $(BUILDROOT)/output/images/rootfs.ext2
 
 RELEASE_BINS := target/riscv64gc-unknown-none-elf/release/
 DEBUG_BINS := target/riscv64gc-unknown-none-elf/debug/
 
 KERNEL_ADDR := 0xc0200000
 INITRD_ADDR := 0xc2200000
 BOOTARGS := keep_bootcon earlycon=sbi
+BUILDROOT_BOOTARGS := root=/dev/nvme0n1 console=hvc0 earlycon=sbi
```
Comment (abrestic-rivos) [inline Makefile:33]: these bootargs should work for debian too, right?

File: Makefile [hunk 9]
```diff
@@ -126,11 +132,34 @@ run_debian: salus
...
 		-append "$(BOOTARGS) root=LABEL=rootfs" \
 		-device guest-loader,initrd=$(INITRD_IMAGE),addr=$(INITRD_ADDR) \
-		-device x-riscv-iommu-pci \
 		-drive file=$(ROOTFS_IMAGE),if=none,id=hd \
-		-device nvme,serial=deadbeef,drive=hd \
-		-netdev user,id=usernet,hostfwd=tcp:127.0.0.1:7722-0.0.0.0:22 \
-		-device e1000e,netdev=usernet \
+		-$(NVME_DEVICE_ARGS) \
+		-$(IOMMU_ARGS) \
+		-$(NETWORK_ARGS) \
```
Comment (abrestic-rivos) [inline Makefile:138]: nit: just put the leading `-` in the definition of the variable

Comment (atishp04) [reply Makefile:138]: Sure.

File: Makefile (hunk 8 above)
Comment (atishp04) [reply Makefile:33]: The console & earlycon part are same. The root part differs as the distro may not have the rootfs on the first partition. I can modify BUILDROOT_BOOTARGS to use $BOOTARGS and have "root=" specific to rootfs type.

File: Makefile (hunk 9 above)
Comment (abrestic-rivos) [reply Makefile:137]: Can you fix this one too? :)

**Review by abrestic-rivos — APPROVED** [review]

**Review by abrestic-rivos — APPROVED** [review]

**Pull request merged**